check_logs.py

```diff
@@ -82,6 +82,13 @@ def verify_build():
         print_red("status.json did not give a pass/fail result!")
         sys.exit(1)
 
+    if build["status_message"] == "Unable to apply kernel patch":
```

Comment:
Looking at [an example](https://github.com/ClangBuiltLinux/continuous-integration2/runs/4566921218?check_suite_focus=true), it looks like when `"status_message"` == "Unable to apply kernel patch"` that `"result" == "fail"`.

Let's say the first request we make has `"status_message"` == "Unable to apply kernel patch"` and `"result" == "fail"`. In such a case, we will retry `max_retries` times, then exit (L83).

I think we need to check this either before the retry loop, or within the retry loop.

Reply:
> Looking at [an example](https://github.com/ClangBuiltLinux/continuous-integration2/runs/4566921218?check_suite_focus=true), it looks like when `"status_message" == "Unable to apply kernel patch"` AND `"result" == "fail"`.

Correct, that should always be what it is.

> Let's say the first request we make has `"status_message" == "Unable to apply kernel patch"` and `"result" == "fail"`. In such a case, we will retry `max_retries` times, then exit (L83).

Shouldn't we bail out of the while loop before getting to `max_retries` because `"result" == "fail"`?

```
    while retries < max_retries:
        if build["result"] == "fail" or build["result"] == "pass":
            break
```

I do not necessarily mind moving it earlier but I don't think there will ever be a time where `"status_message"` is `"Unable to apply kernel patch"` and `"result"` is not `"fail"`.

Reply:
> Shouldn't we bail out of the while loop before getting to max_retries because "result" == "fail"?

Yes, but then we'll exit (L83) BEFORE checking "status_message" in that case. The code you're adding in this PR looks like it will be unreachable to me.

Reply:
ah, no it's not nvm.